packages/lesswrong/components/common/NavigationMenu.jsx

```diff
@@ -7,48 +7,76 @@ import { Link } from 'react-router';
 import { withStyles } from '@material-ui/core/styles';
 
 const styles = theme => ({
-  drawerPaper: {
+  drawerPaperWithoutToC: {
     width: 225,
   },
+  
+  // If there's a table of contents, make the drawer wider than if there isn't
+  // one (because post section titles tend to be longer than top-level
+  // navigation options).
+  drawerPaperWithToC: {
+    width: 300,
+  },
 });
 
-const NavigationMenu = ({open, handleOpen, handleClose, classes}) => {
+const NavigationMenu = ({open, handleOpen, handleClose, classes, toc}) => {
   const af = getSetting('AlignmentForum', false);
   return <SwipeableDrawer
     open={open}
     onClose={(event) => handleClose()}
     onOpen={(event) => handleOpen()}
     classes={{
-      paper: classes.drawerPaper
+      paper: toc ? classes.drawerPaperWithToC : classes.drawerPaperWithoutToC
     }}
   >
-    <MenuItem onClick={handleClose} containerElement={<Link to={"/"}/>}> HOME </MenuItem>
+    <MenuItem onClick={handleClose} containerElement={<Link to={"/"}/>}> Home </MenuItem>
+    
     <Divider />
-    {!af && <MenuItem onClick={handleClose} containerElement={<Link to={"/library"}/>}> LIBRARY </MenuItem>}
+    
+    {toc && <React.Fragment>
+      <Components.TableOfContentsList
+        document={toc.document}
+        sections={toc.sections}
+        onClickSection={() => handleClose()}
+        drawerStyle={true}
+      />
+      <Divider />
+    </React.Fragment>}
+    
+    {!af && <MenuItem onClick={handleClose} containerElement={<Link to={"/library"}/>}> Library </MenuItem>}
+    
     {!af && <MenuItem
       onClick={handleClose}
       innerDivStyle={{paddingLeft:"35px" }}
       containerElement={<Link to={"/rationality"}/>}>
-      RATIONALITY: A-Z
+      Rationality: A-Z
     </MenuItem>}
+    
     {!af && <MenuItem
       onClick={handleClose}
       innerDivStyle={{paddingLeft:"35px" }}
       containerElement={<Link to={"/codex"}/>}>
-      THE CODEX
+      The Codex
```

Comment:
These were intentionally capitalized :P 

We can debate whether they should be, but I think it looks better when they are.

Reply:
I vote for mixed case. I find all-caps actively aversive to read, and I think keeping this all-capsed would keep me (and others) from ever using the drawer.